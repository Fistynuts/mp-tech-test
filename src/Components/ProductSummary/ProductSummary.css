.product-summary--container {
  margin: 4%;
}

.product-summary--image {
  object-fit: contain;
  width: 100%; /* object-fit: contain only works if the image width can be calculated, we want it to be as big as possible */
}
