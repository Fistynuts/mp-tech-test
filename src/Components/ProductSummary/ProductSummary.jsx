import * as React from 'react';
import PropTypes from 'prop-types';

import './ProductSummary.css';

export default class ProductSummary extends React.PureComponent {

  static propTypes = {
    product: PropTypes.objectOf(PropTypes.any).isRequired
  };
  
  render() {
    const { product } = this.props;

    return (
      <div className="product-summary--container">
        <img className="product-summary--image" src={product.ProductImage.Link.Href} alt={product.Title} />
      </div>
    );
  }
}