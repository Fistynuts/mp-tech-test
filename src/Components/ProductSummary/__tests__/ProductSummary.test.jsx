import React from 'react';
import { shallow } from 'enzyme';
import ProductSummary from '../ProductSummary';

describe('ProductSummary', () => {

  const testProduct = {
    Price: {
      Value: 3.29,
      Currency: 'Â£'
    },
    SoldOut: 0,
    Title: 'Champagne Bottle Personalised Congratulations Card',
    ProductCategory: {
      ProductCategoryId: 1,
      Name: 'greeting cards'
    },
    PhotoUploadCount: 0,
    CardShopId: 1,
    DirectSmile: true,
    DefaultSizeId: 1,
    ProductId: 17715,
    MoonpigProductNo: 'll037',
    TradingFaces: 0,
    IsLandscape: 0,
    ShortDescription: null,
    IsCustomisable: 1,
    IsMultipack: 0,
    SeoPath: 'champagne-bottle-personalised-congratulations-card',
    ProductCategoryGroupSeoPath: 'personalised-cards/',
    ProductLink: {
      Href: 'https://api-rest.moonpig.com/v2/products/ll037/1',
      Method: 'GET',
      Rel: 'self',
      Title: '17715'
    },
    ProductImage: {
      Link: {
        Href: 'https://d1xkhapf8f3lxw.cloudfront.net/api/images/CardShop/1/product/ll037',
        Method: 'GET',
        Rel: 'self',
        Title: '17715'
      },
      MimeType: 'image/jpeg'
    },
    Reviews: {
      MinReviewData: 0,
      MaxReviewData: 0,
      AverageStarReviewRating: 4.7561,
      ReviewCount: 41
    }
  };

  it('should match the snapshot', () => {
    const app = shallow(<ProductSummary product={testProduct} />);
    expect(app).toMatchSnapshot();
  });
});
