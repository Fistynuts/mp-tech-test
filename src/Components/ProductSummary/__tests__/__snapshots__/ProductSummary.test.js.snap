// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSummary matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductSummary
    product={
      Object {
        "CardShopId": 1,
        "DefaultSizeId": 1,
        "DirectSmile": true,
        "IsCustomisable": 1,
        "IsLandscape": 0,
        "IsMultipack": 0,
        "MoonpigProductNo": "ll037",
        "PhotoUploadCount": 0,
        "Price": Object {
          "Currency": "Â£",
          "Value": 3.29,
        },
        "ProductCategory": Object {
          "Name": "greeting cards",
          "ProductCategoryId": 1,
        },
        "ProductCategoryGroupSeoPath": "personalised-cards/",
        "ProductId": 17715,
        "ProductImage": Object {
          "Link": Object {
            "Href": "https://d1xkhapf8f3lxw.cloudfront.net/api/images/CardShop/1/product/ll037",
            "Method": "GET",
            "Rel": "self",
            "Title": "17715",
          },
          "MimeType": "image/jpeg",
        },
        "ProductLink": Object {
          "Href": "https://api-rest.moonpig.com/v2/products/ll037/1",
          "Method": "GET",
          "Rel": "self",
          "Title": "17715",
        },
        "Reviews": Object {
          "AverageStarReviewRating": 4.7561,
          "MaxReviewData": 0,
          "MinReviewData": 0,
          "ReviewCount": 41,
        },
        "SeoPath": "champagne-bottle-personalised-congratulations-card",
        "ShortDescription": null,
        "SoldOut": 0,
        "Title": "Champagne Bottle Personalised Congratulations Card",
        "TradingFaces": 0,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="Champagne Bottle Personalised Congratulations Card"
        className="product-summary--image"
        src="https://d1xkhapf8f3lxw.cloudfront.net/api/images/CardShop/1/product/ll037"
      />,
      "className": "product-summary--container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "Champagne Bottle Personalised Congratulations Card",
        "className": "product-summary--image",
        "src": "https://d1xkhapf8f3lxw.cloudfront.net/api/images/CardShop/1/product/ll037",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="Champagne Bottle Personalised Congratulations Card"
          className="product-summary--image"
          src="https://d1xkhapf8f3lxw.cloudfront.net/api/images/CardShop/1/product/ll037"
        />,
        "className": "product-summary--container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Champagne Bottle Personalised Congratulations Card",
          "className": "product-summary--image",
          "src": "https://d1xkhapf8f3lxw.cloudfront.net/api/images/CardShop/1/product/ll037",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
